vcsclient/github.go

```diff
@@ -336,27 +336,53 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
 		}
 	}
 
-	sourceBranch, err1 := extractBranchFromLabel(vcsutils.DefaultIfNotNil(pullRequest.Head.Label))
-	targetBranch, err2 := extractBranchFromLabel(vcsutils.DefaultIfNotNil(pullRequest.Base.Label))
-	err = errors.Join(err1, err2)
-	if err != nil {
-		return PullRequestInfo{}, err
+	return mapGitHubPullRequestToPullRequestInfo(pullRequest, false)
+}
+
+func mapGitHubPullRequestToPullRequestInfo(ghPullRequest *github.PullRequest, withBody bool) (PullRequestInfo, error) {
+	var sourceBranch, targetBranch string
+	var err1, err2 error
+	if ghPullRequest != nil && ghPullRequest.Head != nil && ghPullRequest.Base != nil {
```

Comment:
why?

Reply:
Head and Base are pointers, could have a nil values

Reply:
@sverdlov93